shell/platform/fuchsia/flutter/flatland_external_view_embedder.cc

```diff
@@ -14,14 +14,19 @@ FlatlandExternalViewEmbedder::FlatlandExternalViewEmbedder(
     std::string debug_label,
     fuchsia::ui::views::ViewCreationToken view_creation_token,
     scenic::ViewRefPair view_ref_pair,
+    fuchsia::ui::composition::ViewBoundProtocols endpoints,
     fidl::InterfaceRequest<fuchsia::ui::composition::ParentViewportWatcher>
         parent_viewport_watcher_request,
     FlatlandConnection& flatland,
     SurfaceProducer& surface_producer,
     bool intercept_all_input)
     : flatland_(flatland), surface_producer_(surface_producer) {
-  flatland_.flatland()->CreateView(std::move(view_creation_token),
-                                   std::move(parent_viewport_watcher_request));
+  fuchsia::ui::views::ViewIdentityOnCreation view_identity = {
```

Comment:
What's the plan for this one vs scenic::ViewRefPair?

I'd prefer the FIDL type long-term, but would we still want a factory function to make generating the pair easier?

Also, does the FIDL type allow strucutred binding like auto [view_ref, view_ref_control]?  It should, based on your use of brace initialization here.

Reply:
ViewRefPair for legacy usages only - documented as such already.

https://fuchsia.googlesource.com/fuchsia/+/refs/heads/main/sdk/lib/ui/scenic/cpp/view_ref_pair.h#14

And yes, we have a factory function, I'll use that. Forgot about it.

Reply:
Actually the view ref pair is piped in from outside, and enters through Engine() ctor. Given that "view token pair" also enters the same way, I moved this up to the Engine callback that constructs the FlatlandExternalViewEmbedder itself.